pilot/pkg/model/gateway.go

```diff
@@ -95,14 +95,18 @@ func MergeGateways(gateways ...Config) *MergedGateway {
 						// WE cannot merge two HTTPS servers even if their TLS settings have same path to the keys, because we don't know if the contents
 						// of the keys are same. So we treat them as effectively different TLS settings.
 						if _, exists := rdsRouteConfigNames[rdsName]; exists {
-							log.Debugf("skipping server on gateway %s port %s.%d.%s: non unique port name for HTTPS port",
+							log.Infof("skipping server on gateway %s port %s.%d.%s: non unique port name for HTTPS port",
 								spec.Name, s.Port.Name, s.Port.Number, s.Port.Protocol)
 							continue
 						}
 						rdsRouteConfigNames[rdsName] = []*networking.Server{s}
 					}
 
 					// We have another TLS server on the same port. Can differentiate servers using SNI
+					if s.Tls == nil {
+						log.Warnf("TLS server without TLS options %s %s", name, s.String())
+						continue
+					}
```

Comment:
Please remove this. I have added validation to ensure that HTTPS/TLS servers should have TLS settings and plain text HTTP servers cannot have any TLS settings.

Reply:
https://github.com/istio/istio/pull/7175/files#diff-5a64dc2088020f89bc6d77354b9ebb22R414 .. Will ensure that you cannot have a TLS server with non TLS port, and vice versa.

Reply:
Got it, thanks - will merge your PR and remove this. Forgot to check - do you warn with enough info or have a counter ? I found it very hard to use the errors in pilot without having the namespace and name of the config.